contracts/core/extensions/CoreAccounting.sol

```diff
@@ -211,30 +211,30 @@ contract CoreAccounting is
         // Declare interface variavle for vault
         IVault vault = IVault(state.vault);
 
-        // Call Vault contract to deattribute tokens to user
-        vault.decrementTokenOwner(
+        // Call Vault contract to withdraw tokens from Vault to user
+        vault.withdrawTo(
```

Comment:
What was the thinking behind this? We have it in the existing order for a reason, we update the state before we perform the actual transfer

Reply:
Rule of thumb

Reply:
Just changed it back, I see the `require` in `decrementTokenOwner` that needs to happen first. I originally changed it to mirror deposit where the transfer happens first and then the increment.